src/features/movies/MovieList/popularMoviesSaga.js

```diff
@@ -1,22 +1,37 @@
 import { delay, call, put, takeLatest } from "redux-saga/effects";
-import { APIgenresUrl, APIPopularMoviesUrl } from "../../APIdata";
+import {
+  APIgenresUrl,
+  APIPopularMoviesUrl,
+  APISearchMoviesUrl,
+} from "../../APIdata";
 import { getAPI } from "../../getAPI";
-import { fetchMoviesError, fetchMovies, fetchMoviesSuccess, setGenres } from "./popularMoviesSlice";
+import {
+  fetchMovies,
+  fetchMoviesSuccess,
+  fetchMoviesError,
+  setGenres,
+} from "./popularMoviesSlice";
 
-function* fetchPopularMoviesWorker() {
+function* fetchPopularMoviesWorker({ payload: { query } }) {
   const popularMoviesPage1 = `${APIPopularMoviesUrl}&page=1`;
+  const searchMovies = `${APISearchMoviesUrl}${query}`;
 
   try {
     yield delay(1000);
-    const popularMovies = yield call(getAPI, popularMoviesPage1);
+    const request = yield call(
+      getAPI,
+      !query 
+        ? popularMoviesPage1 
+        : searchMovies
```

Comment:
I'd take it outside the try block to a constant

Reply:
Great idea!